lib/node_modules/@stdlib/array/base/cusome/lib/assign.js

```diff
@@ -0,0 +1,312 @@
+	}
+	return ydata;
+}
+
+/**
+* Cumulatively tests whether at least n array elements in a provided array are truthy.
+*
+* @private
+* @param {Collection} x - input array
+* @param {integer} n - number of elements to be truthy
+* @param {Object} y - output array object
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @returns {Collection} output array
+*
+* @example
+* var Uint8Array = require( '@stdlib/array/uint8' );
+* var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+*
+* var x = new Uint8Array( [ 0, 0, 0, 1, 0 ] );
+* var n = 2;
+* var y = toAccessorArray( [ false, null, false, null, false, null, false, null, false, null ] );
+*
+* var arr = boolean( x, n, arraylike2object( y ), 2, 0 );
+*
+* var v = y.get( 0 );
+* // returns false
+*
+* v = y.get( 2 );
+* // returns false
+*
+* v = y.get( 4 );
+* // returns false
+*
+* v = y.get( 6 );
+* // returns false
+*
+* v = y.get( 8 );
+* // returns false
+*/
+function boolean( x, n, y, stride, offset ) {
+	var ydata;
+	var yset;
+	var flg;
+	var io;
+	var i;
+
+	yset = y.accessors[ 1 ];
+	ydata = y.data;
+
+	flg = false;
+	io = offset;
+	for ( i = 0; i < x.length; i++ ) {
+		if ( flg === false && n > 0 && x[ i ] ) {
+			n -= 1;
+		}
+		if (n === 0) {
+			flg = true;
+		}
+		yset( ydata, io, flg );
+		io += stride;
+	}
+	return ydata;
+}
+
+
+// MAIN //
+
+/**
+* Cumulatively tests whether at least n array elements in a provided array are truthy and assigns results to provided output array.
+*
+* @param {Collection} x - input array
+* @param {integer} n - number of elements to be truthy
+* @param {Collection} y - output array
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @returns {Collection} output array
+*
+* @example
+* var x = [ false, false, false, true, true ];
+* var n = 2;
+* var y =  [ false, null, false, null, false, null, false, null, false, null ];
+*
+* var out = assign( x, n, y, 2, 0 );
+* // returns [ false, null, false, null, false, null, false, null, true, null ]
+*
+* var bool = ( y === out );
+* // returns true
+*/
+function assign( x, n, y, stride, offset ) {
+	var xo = arraylike2object( x );
+	var yo = arraylike2object( y );
+	if (
+		xo.accessorProtocol ||
+		yo.accessorProtocol
+	) {
+		// If provided a complex number array, reinterpret as a real typed array and test interleaved real and imaginary components, where we consider a complex number to be truthy if at least one component is non-zero...
+		if ( isComplex128Array( x ) ) {
+			complex( reinterpret128( x, 0 ), n, yo, stride, offset );
+		} else if ( isComplex64Array( x ) ) {
+			complex( reinterpret64( x, 0 ), n, yo, stride, offset);
+		} else if ( isBooleanArray( x ) ) {
+			boolean( reinterpretBoolean( x, 0 ), n, yo, stride, offset);
+		} else {
+			accessors( xo, n, yo, stride, offset );
+		}
```

Comment:
```suggestion
			return complex( reinterpret128( x, 0 ), n, yo, stride, offset );
		}
		if ( isComplex64Array( x ) ) {
			return complex( reinterpret64( x, 0 ), n, yo, stride, offset);
		}
		if ( isBooleanArray( x ) ) {
			return boolean( reinterpretBoolean( x, 0 ), n, yo, stride, offset);
		}
		 accessors( xo, n, yo, stride, offset );
```

Prefer early returns.